qiskit/pulse/builder.py

```diff
@@ -0,0 +1,1986 @@
+
+    Examples:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+
+        d0 = pulse.DriveChannel(0)
+        d1 = pulse.DriveChannel(1)
+
+        with pulse.build() as pulse_prog:
+            with pulse.pad():
+                with pulse.align_right():
+                    # this pulse will start at t=0
+                    pulse.play(pulse.Constant(100, 1.0), d0)
+                    # this pulse will start at t=80
+                    # a delay will be inserted from t=0 to t=80
+                    pulse.play(pulse.Constant(20, 1.0), d1)
+        assert pulse_prog.ch_start_time(d0) == pulse_prog.ch_start_time(d1)
+        assert pulse_prog.ch_stop_time(d0) == pulse_prog.ch_stop_time(d1)
+    """
+
+
+@contextmanager
+def transpiler_settings(**settings) -> ContextManager[None]:
+    """Set the currently active transpiler settings for this context.
+
+    Examples:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+        from qiskit.test.mock import FakeOpenPulse2Q
+
+        backend = FakeOpenPulse2Q()
+
+        with pulse.build(backend):
+            print(pulse.active_transpiler_settings())
+            with pulse.transpiler_settings(optimization_level=3):
+                print(pulse.active_transpiler_settings())
+    """
+    builder = _active_builder()
+    curr_transpiler_settings = builder.transpiler_settings
+    builder.transpiler_settings = collections.ChainMap(
+        settings, curr_transpiler_settings)
+    try:
+        yield
+    finally:
+        builder.transpiler_settings = curr_transpiler_settings
+
+
+@contextmanager
+def circuit_scheduler_settings(**settings) -> ContextManager[None]:
+    """Set the currently active circuit scheduler settings for this context.
+
+    Examples:
+
+    .. jupyter-execute::
+
+        from qiskit import pulse
+        from qiskit.test.mock import FakeOpenPulse2Q
+
+        backend = FakeOpenPulse2Q()
+
+        with pulse.build(backend):
+            print(pulse.active_circuit_scheduler_settings())
+            with pulse.circuit_scheduler_settings(method='alap'):
+                print(pulse.active_circuit_scheduler_settings())
+    """
+    builder = _active_builder()
+    curr_circuit_scheduler_settings = builder.circuit_scheduler_settings
+    builder.circuit_scheduler_settings = collections.ChainMap(
+        settings, curr_circuit_scheduler_settings)
+    try:
+        yield
+    finally:
+        builder.circuit_scheduler_settings = curr_circuit_scheduler_settings
+
+
+@contextmanager
+def phase_offset(phase: float,
+                 channel: channels.PulseChannel
+                 ) -> ContextManager[None]:
+    """Shift the phase of a channel on entry into context and undo on exit.
+
+    Examples:
+
+    .. jupyter-execute::
+
+        import math
+
+        from qiskit import pulse
+
+        d0 = pulse.DriveChannel(0)
+
+        with pulse.build(backend) as pulse_prog:
+            with pulse.phase_offset(math.pi, d0):
```

Comment:
is it possible to take multiple channels to sync control channels?

Reply:
Sure, this makes sense. Added in 9f0d2540d.